server/liveblog/posts/posts.py

```diff
@@ -420,12 +420,45 @@ def on_update(self, updates, original):
             content_diff = True
         else:
             for index, val in enumerate(updates["groups"][1]["refs"]):
-                item = get_resource_service("archive").find_one(
-                    req=None, _id=val["residRef"]
-                )
-                if item["text"] != original["groups"][1]["refs"][index]["item"]["text"]:
-                    content_diff = True
-                    break
+                service_name = val.get("location", "archive")
+                service = get_resource_service(service_name)
+                item = service.find_one(req=None, _id=val["residRef"])
+                item_type = item.get("item_type")
+
+                if item_type == "poll":
+                    item_poll_body = item.get("poll_body", {})
+                    original_poll_body = original["groups"][1]["refs"][index][
```

Comment:
this piece of code is used multiple times and it seems to me that we get it from the `enumerate`. Do you know what's the reason to get it like again and again? See line `458` too. 
`original["groups"][1]["refs"][index]`
If it's not possible to use the `item_ref` suggested above from the `enumerate`, perhaps we should extract it once and reuse the variable, I think that would make the code more readable.

Reply:
Another thing is that all these code from line `415` to `461` is to detect if the content has changed. Maybe we should extract it into a separate function and cover it with tests. What do you think?

Reply:
I have extracted the section into a function called `check_content_diff`. 

Also note that I have simplified the check for polls to just check the `active_until` value since that's the only value that can be changed from the Editor's end before the poll elapses. Hence the question and answers aren't checked.